roles/replace_node/tasks/volume.yml

```diff
@@ -0,0 +1,113 @@
+---
+# Set up the volume management
+- name: Fetch the directory and volume details
+  block:
+    - name: Get the list of volumes on the machine
+      shell: ls "{{ glusterd_libdir }}/vols"
+      register: dir_list
+
+    - set_fact:
+        volumes: "{{ dir_list.stdout.split() }}"
+
+    # Find the list of bricks on the machine
+    - name: Get the list of bricks corresponding to volume
+      shell: >
+        gluster vol info {{ item }} | grep "Brick.*{{ item }}:" |
+        awk -F: '{ print $3 }'
+      with_items: "{{ volumes }}"
+      register: brick_list
+
+    - set_fact:
+        volume_bricks: {}
+
+    - name: Get the list of bricks into a variable
+      set_fact:
+        volume_bricks: "{{ volume_bricks|combine({item.item: item.stdout}) }}"
+      loop: "{{ brick_list.results }}"
+  delegate_to: "{{ gluster_maintenance_cluster_node }}"
+
+- name: Create the brick directory if not already present
```

Comment:
Shouldn't it error out if brick is not mounted? Otherwise the brick would end up being created on root partition?

Reply:
In that case reset-brick would fail. No?

However, we have to ask for another input parameter from user. For if the gluster-brick directory retrieved is /data/disk/foo, we do not know if /data is mountpoint or /data/disk is... 

What do you suggest?

Reply:
@sabose I think I was not clear, I meant we do not have enough data to check if the brick directory has a mount point. When I said reset-brick would fail, I meant the next step will fail anyway. 

I think we should has for the mountpoint as an input parameter from the user. And check if that is actually mounted. I will think through this.

Reply:
The brick path should be same as the one returned in volume info - so additional input from user is probably not needed. We also do not know the number of bricks on the node that will be replaced

Reply:
We know the brick path, what we don't know is the mount point. For eg: if brick path is: /data/foo/bar/baz we do not know if /data is the mount point or /data/foo or /data/foo/bar ... 

We have to check if the brick path is on a mounted filesystem or on / ... since we do not know the mount point it is tricky.

Reply:
What I meant is - get rid of this step - i.e creating directory. Either we do the entire process of provisioning and mounting brick, or assume that the user has already done this